tests/Calculator/CalculatorTest.cpp

```diff
@@ -0,0 +1,153 @@
+//
+// Created by Yasmine Farah on 07/11/2024.
+//
+#include "gtest/gtest.h"
+#include "../../src/Calculator/Calculator.h"
+#include "../../src/Calculator/LennardJonesCalculator.h"
+#include "../../src/Objects/ParticleContainer.h"
+#include "../../src/Objects/Particle.h"
+
+/*Checks that calculateX() correctly updates the position of a particle*/
+TEST(CalculatorTest, correctXcalculations) {
+    SPDLOG_DEBUG("running test");
+    ParticleContainer test;
+    SPDLOG_DEBUG("created container");
+    Particle p({0.0, 0.0, 0.0},{0.1, 0.1, 0.1},1.0,0);
+    SPDLOG_DEBUG("created particle");
+    p.setF({0.2, 0.2, 0.2});
+    test.addParticle(p);
+    Calculators::Calculator calc;
+    calc.calculateX(test,0.5); //value different from 1 to check that (delta_t)^2 is calculated correctly
+
+    for (auto &t : test) {
+        SPDLOG_DEBUG("Looking at particle");
+        EXPECT_NEAR(0.075, t.getX()[0], 0.00001);
+        EXPECT_NEAR(0.075, t.getX()[1], 0.00001);
+        EXPECT_NEAR(0.075, t.getX()[2], 0.00001);
+    }
+
+}
+
+/*Checks that calculateF() correctly updates the force between particles*/
+TEST(CalculatorTest, correctFIJcalculations) {
+    ParticleContainer test;
+    std::array<double,3> old = {1.0, 2.0, 3.0};
+
+    Particle i({0.0, 0.0, 0.0},{0.0, 0.0, 0.0},1.0,0);
+    i.setF(old);
+    test.addParticle(i);
+
+    Particle j({2.0, 2.0, 2.0},{0.0, 0.0, 0.0},2.0,0);
+    j.setF(old);
+    test.addParticle(j);
+
+    Particle k({4.0, 4.0, 4.0},{0.0, 0.0, 0.0},3.0,0);
+    k.setF(old);
+    test.addParticle(k);
+
+    test.initializePairsVector();
+
+
+    Calculators::Calculator calc;
+    calc.calculateF(test); 
+
+    for (Particle& p: test) {
+        ASSERT_EQ(old, test.getParticle(p.getID()).getOldF());
```

Comment:
Why use `ASSERT` here? I recommend using `EXPECT` unless there is a reason to abort the test. See https://testing.googleblog.com/2008/07/tott-expect-vs-assert.html